### PR TITLE
Add background location update and emulate waypoints

### PR DESCRIPTION
**Pull request by lukasz-szyszkowski**
This branch is created on top of the `feature/move-xcworkspace` and won't pass GH Actions (see info here https://github.com/ably/ably-asset-tracking-swift/pull/199)

### CODE REVIEW

**Comment by lukasz-szyszkowski** [comment]
Demo video

https://user-images.githubusercontent.com/47628502/141738598-26d49e2d-9aef-40d9-8690-c1ab548b6602.mp4

**Comment by lukasz-szyszkowski** [comment]
Demo video

https://user-images.githubusercontent.com/47628502/141739311-b07e78e3-2579-438c-a57d-57435e1132ef.mov

**Review by ben-xD — APPROVED** [review]

File: Sources/AblyAssetTrackingCore/Configuration/ConnectionConfiguration.swift [hunk 1]
```diff
@@ -10,6 +10,30 @@ public enum AuthResult {
     case tokenDetails(TokenDetails)
 }
 
+public class ObjcAuthResultJWT: NSObject {
```
Comment (KacperKluka) [inline Sources/AblyAssetTrackingCore/Configuration/ConnectionConfiguration.swift:13]: Should this class and the two below classes be part of this PR? :thinking:

**Review by KacperKluka — APPROVED** [review]
Approving based on the [video you've provided](https://github.com/ably/ably-asset-tracking-swift/pull/205#issuecomment-968606233) as I don't have much iOS knowledge :wink:

**Comment by lawrence-forooghian** [comment]
@lukasz-szyszkowski the `feature/move-xcworkspace` branch was already merged in https://github.com/ably/ably-asset-tracking-swift/pull/212, but the diff for this PR still contains a bunch of changes that seem related to that work. This makes it hard to review. Please could you take a look?

**Comment by lukasz-szyszkowski** [comment]
> I'm finding it quite hard to review this PR. It seems that not all of the commits are relevant to the title, perhaps some of them are leftovers from other branches that have now been merged?
> 
> I'd appreciate it if you could perhaps either open a new PR with only the relevant commits, or rebase this branch on `main` so that the commit history is more reviewable.

here #242

**Pull request closed without merging**